### ISSUE TITLE
[Bug][1.16.5] Crystalizer consumes twice the desired fluid per recipe, and outputting with improper inputs

### ISSUE THREAD

**Issue by AubriTheHuman** [Immersive-Geology-Team/Immersive-Geology#134]
IE version: 1.16.5-5.1.0-148
IG version: Beta v1.1 MC 1.16.5

When making a custom recipe for the crystallizer, it will consume twice the desired amount of fluid. so 250mB -> 500mB, 500mB -> 1000mB. While inherently not an issue, it is a problem with a recipe needing 125mB of fluid.

On top of this, feeding less than the necessary amount of fluid into the crystallizer will cause it to craft, consuming the fluid and outputting a product, despite only consuming 150mB. None of this behavior makes any output logs at all.

Custom Recipe:
```json
{
    "type": "immersive_geology:crystalizer",
    "result": {
      "item": "malum:aquatic_spirit"
    },
    "fluid_input": {
      "tag": "forge:distillations/aquatic_spirit_distillation",
      "amount": 250
    },
    "time": 400,
    "energy": 25600
}
```

This will consume 500mB of distillation, and output 1 item. And will do this behavior when the crystallizer has 2000mB of fluid within it, so it does buffer. But I'd expect 8 products instead. On top of this, as said before, feeding it 50mB of the fluid from an IE tank and pipes will cause it to craft, even with a fully empty buffer (ignoring the amount required). Feeding it 600mB produces 2 products and consumes the 600 mB. 

Imgur album of block data during use:  [https://imgur.com/a/ycD2R3L](https://imgur.com/a/ycD2R3L)

(one thing I just noticed is that despite extracting the item with an IE extractor conveyor belt, the data still shows it containing items in the output buffer, yet aren't extracting, but are at the same time? if inputting a different fluid it prevents any output as a result. you can see this in the Imgur album)

**Comment by muddykat-tech** [comment]
thank you very much, this issue will be addressed when possible.

**Comment by AubriTheHuman** [comment]
This is technically 3 bug reports at once, so I hope that's not an issue.

**Comment by UnSchtalch** [comment]
Fixed in [2dd7761](https://github.com/Immersive-Geology-Team/Immersive-Geology/commit/2dd776170d2222033e8507386d6c615c3519a0bc)
